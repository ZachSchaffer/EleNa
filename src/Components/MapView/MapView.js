import React, { Component } from 'react';
import ReactMapGL, { Marker } from 'react-map-gl';
import RoomIcon from '@material-ui/icons/Room';
import PropTypes from 'prop-types';

class MapView extends Component {
  constructor(props) {
    super(props);
    this.viewport = {
      width: '70vw',
      height: '80vh',
      latitude: 42.380368,
      longitude: -72.523143,
      zoom: 12,
    };
  }

  render() {
    return (
      <ReactMapGL
        {...this.viewport}
        onViewportChange={(viewport) => this.setState({ viewport })}
        mapStyle={`https://api.maptiler.com/maps/streets/style.json?key=${process.env.REACT_APP_MAPTILLER_API_KEY}`}
        mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_API_KEY}
      >
        {this.props.markers.map(
          (marker) =>
            marker && (
              <Marker
                key={marker.getLatitude()}
                latitude={marker.getLatitude()}
                longitude={marker.getLongitude()}
                offsetLeft={-20}
                offsetTop={-10}
              >
                <RoomIcon color="error" />
              </Marker>
            )
        )}
      </ReactMapGL>
    );
  }
}

MapView.propTypes = {
  markers: PropTypes.array,
};

export default MapView;
