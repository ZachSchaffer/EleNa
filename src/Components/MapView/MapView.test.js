import React from 'react';
import renderer from 'react-test-renderer';
import MapView from './MapView';

test('test MapView', () => {
    const tree = renderer.create(<MapView markers = {[]}/>).toJSON();
    expect(tree).toMatchSnapshot();
});