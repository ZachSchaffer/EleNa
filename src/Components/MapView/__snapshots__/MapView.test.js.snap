// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MapView 1`] = `
<div
  style={
    Object {
      "cursor": "grab",
      "height": 0,
      "position": "relative",
      "width": 0,
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      className=""
      style={
        Object {
          "height": "100%",
          "overflow": "hidden",
          "position": "absolute",
          "visibility": "inherit",
          "width": "100%",
        }
      }
    />
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    />
  </div>
</div>
`;
